addons/account/models/account_move.py

```diff
@@ -2048,6 +2049,106 @@ def _validate_taxes_country(self):
                     raise ValidationError(_("This entry contains taxes that are not compatible with your fiscal position. Check the country set in fiscal position and in your tax configuration."))
                 raise ValidationError(_("This entry contains one or more taxes that are incompatible with your fiscal country. Check company fiscal country in the settings and tax country in taxes configuration."))
 
+    # -------------------------------------------------------------------------
+    # CATALOG
+    # -------------------------------------------------------------------------
+    def _get_action_add_from_catalog_extra_context(self):
+        res = super()._get_action_add_from_catalog_extra_context()
+        if not self.is_sale_document() and self.move_type != 'entry' and self.partner_id:
+            res['search_default_seller_ids'] = self.partner_id.name
+
+        res['product_catalog_currency_id'] = self.currency_id.id
+        res['product_catalog_digits'] = self.line_ids._fields['price_unit'].get_digits(self.env)
+        return res
+
+    def _get_product_catalog_domain(self):
+        if self.is_sale_document():
+            return expression.AND([super()._get_product_catalog_domain(), [('sale_ok', '=', True)]])
+        elif self.is_purchase_document():
+            return expression.AND([super()._get_product_catalog_domain(), [('purchase_ok', '=', True)]])
+        else: # In case of an entry
```

Comment:
```suggestion
        else:  # In case of an entry
```
Cf: [PEP8](https://peps.python.org/pep-0008/#:~:text=Inline%20comments%20should%20be%20separated,a%20%23%20and%20a%20single%20space.)
:nerd_face: